quic/s2n-quic-core/src/packet/interceptor.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{event::api::Subject, packet::number::PacketNumber, time::Timestamp};
+use s2n_codec::{DecoderBufferMut, EncoderBuffer};
+
+#[derive(Debug)]
+pub struct Packet {
```

Comment:
should this be `non_exhaustive`?

Reply:
I think it can be once/if the feature is stabilized. Until then I think we can get away with it not being backward compatible.

Reply:
Maybe put a TODO then somewhere so we don't forget

Reply:
That works